src/test/kotlin/XodusTest.kt

```diff
@@ -287,6 +287,65 @@ class XodusTest {
         assertNotNull(agentStore.findById<MapField>("id3") != null)
     }
 
+    @Test
+    fun `should correctly create,update and get an object with id and StreamSerialization annotation`() = runBlocking {
+        val id = CompositeId("one", 1)
+        updateObject(StreamSerializationTestObject(id, listOf("one", "two", "three")), id)
+        updateObject(StreamSerializationTestObject(id, listOf("three", "two", "one")), id)
+    }
+
+    @Test
+    fun `should correctly delete an object with StreamSerialization annotation in inner object`() = runBlocking {
+        val id = CompositeId("one", 1)
+        val mapInMapWrapper = MapInMapWrapper(id, mutableMapOf())
+        for (i in 1..5) {
+            val a = MapWrapper(i, mutableMapOf())
+            for (j in 1..5) {
+                a.secondMap["$j"] = TestClass("$j", j)
+            }
+            mapInMapWrapper.map["$i"] = a
+        }
+        agentStore.store(mapInMapWrapper)
+        assertEquals(mapInMapWrapper, agentStore.findById<MapInMapWrapper>(id))
+        agentStore.deleteById<MapInMapWrapper>(id)
+        assertTrue(File("${agentStore.location}\\MapInMapWrapper\\map\\map\\value").listFiles().isNullOrEmpty())
+    }
+
+    @Test
+    fun `shouldn't not delete files belonging to another entity`() = runBlocking {
+        val firstId = CompositeId("one", 1)
+        val secondId = CompositeId("two", 2)
+        val mapInMapWrapper = MapInMapWrapper(firstId, mutableMapOf())
+        val mapInMapWrapper2 = MapInMapWrapper(secondId, mutableMapOf())
+        for (i in 1..5) {
+            val a = MapWrapper(i, mutableMapOf())
+            for (j in 1..5) {
+                a.secondMap["$j"] = TestClass("$j", j)
+            }
+            mapInMapWrapper.map["$i"] = a
+            mapInMapWrapper2.map["$i$i"] = a
+        }
+        agentStore.store(mapInMapWrapper)
+        agentStore.store(mapInMapWrapper2)
```

Comment:
can we do it that way?

```suggestion
        val firstId = CompositeId("one", 1)
        val mapInMapWrapper = storeMapInMapWrapper(firstId)
        val secondId = CompositeId("two", 2)
        val mapInMapWrapper2 = storeMapInMapWrapper(secondId)
```

```
    private suspend fun storeMapInMapWrapper(id: CompositeId): MapInMapWrapper {
        val mapInMapWrapper = MapInMapWrapper(id, mutableMapOf())
        for (i in 1..5) {
            val a = MapWrapper(i, mutableMapOf())
            for (j in 1..5) {
                a.secondMap["$j"] = TestClass("$j", j)
            }
            mapInMapWrapper.map["$i"] = a
        }
        agentStore.store(mapInMapWrapper)
        return mapInMapWrapper
    }
```